db.py

```diff
@@ -6,7 +6,13 @@
 #TODO delete excel
 
 
+# Any function can throw sqlite3.OperationalError if database is locked for too long
 class DB(object):
+    # temporary custom exception until error handling is well defined
+    class DBException(Exception):
```

Comment:
No need to override init. exception subclassing is generally done as
class MyException(Exception): pass
Unless you have something interesting to add in the child class